ci_framework/roles/devscripts/tasks/main.yml

```diff
@@ -78,26 +78,36 @@
     - "{{ cifmw_devscripts_logs_dir }}"
     - "{{ cifmw_devscripts_output_dir }}"
 
-- name: Prepare the host for execution of dev-scripts.
+- name: Preform pre-checks
   tags:
-    - devscripts_prepare
-    - devscripts_deploy
-  ansible.builtin.import_tasks: 01_prepare_host.yml
+    - always
+  ansible.builtin.import_tasks: 00_pre_check.yml
 
-- name: Gather the host and dev-scripts required information.
+- name: Prepare for OCP deployment.
   tags:
     - devscripts_prepare
     - devscripts_deploy
-  ansible.builtin.import_tasks: 02_gather_env_details.yml
+  when: not cifmw_devscripts_skip_deploy | default(false)
+  block:
+    - name: Prepare the host for execution of dev-scripts.
+      ansible.builtin.import_tasks: 01_prepare_host.yml
+
+    - name: Gather the host and dev-scripts required information.
+      ansible.builtin.import_tasks: 02_gather_env_details.yml
 
 - name: Running dev-scripts.
   tags:
     - devscripts_deploy
+  when:
+    - cifmw_devscripts_skip_deploy is defined
+    - not cifmw_devscripts_skip_deploy | bool
   ansible.builtin.import_tasks: 03_install.yml
 
 - name: Prepare for image overlay
   when:
     - cifmw_devscripts_use_layers | bool
+    - cifmw_devscripts_skip_deploy is defined
```

Comment:
i'm wondering if undefined skip_deploy should result in skipping overlays?

Reply:
Thank you

Yes, it should not be skipped when `cifmw_devscripts_skip_deploy` is undefined. Fixed in the latest commit.